src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -9462,6 +9462,59 @@ public void TypePattern_06()
             EOF();
         }
 
+        [Fact, WorkItem(49354, "https://github.com/dotnet/roslyn/issues/49354")]
+        public void TypePattern_07()
+        {
+            UsingStatement(@"_ = e is (int) or string;",
```

Comment:
Consider adding tests where
1. the parenthesized type is not built in, e.g. `_ = e is (Widget) or string;`
2. a constant value is cast to some non-built-in type, e.g. `_ = e is (Widget)42 or string;`

Reply:
Done.